v2/blockstore/readonly.go

```diff
@@ -165,7 +165,7 @@ func (b *ReadOnly) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
 
 	ch := make(chan cid.Cid, 5)
 	go func() {
-		done := ctx.Done()
+		defer close(ch)
```

Comment:
I'd suggest writing out a log line when there's an error (lines 174 and 178)

Reply:
This is a library though, so logging requires some form of API to allow being configured - and to probably be off by default. Is there a standard across PL? I know https://github.com/ipfs/go-log, but that feels too heavy to pull into relatively low-level IPLD libraries.

Maybe I just expose an option like:

```
WithLogging(func(format string, args ...interface{})) Option
```

Reply:
I guess really the error should be returned to the caller somehow but I assume it's not possible to change the AllKeysChan API at this stage.
Suggest we merge this PR as a strict improvement and discuss options in an issue.

Reply:
This is the blockstore interface, so we indeed can't change the method signature :) I'll raise an issue about logging.

Reply:
https://github.com/ipld/go-car/issues/112